File: 217. Contains Duplicate/217.ContainsDuplicate.js
var containsDuplicate = function (nums) {
  let set = new Set(); // Create a Set to store unique elements

  // Loop through each element in the array
  for (let i of nums) {
    // If the element is already in the Set, it means we have found a duplicate
    if (set.has(i)) {
      return true; // Return true because a duplicate is found
    }
    // Otherwise, add the element to the Set to keep track of it
    set.add(i);
  }

  // If the loop completes and no duplicates are found, return false
  return false;
};

File: 217. Contains Duplicate/217.ContainsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()  # Create a Set to store unique elements
        # Loop through each element in the list
        for num in nums:
            # If the element is already in the Set, it means we have found a duplicate
            if num in seen:
                return True  # Return True because a duplicate is found
            # Otherwise, add the element to the Set to keep track of it
            seen.add(num)
        # If the loop completes and no duplicates are found, return False
        return False